pkg/fanal/artifact/image/image.go

```diff
@@ -335,33 +335,38 @@ func (a Artifact) diffIDs(configFile *v1.ConfigFile) []string {
 	})
 }
 
-func (a Artifact) uncompressedLayer(diffID string) (string, io.ReadCloser, error) {
+func (a Artifact) uncompressedLayer(diffID string) (string, int64, io.ReadCloser, error) {
 	// diffID is a hash of the uncompressed layer
 	h, err := v1.NewHash(diffID)
 	if err != nil {
-		return "", nil, xerrors.Errorf("invalid layer ID (%s): %w", diffID, err)
+		return "", 0, nil, xerrors.Errorf("invalid layer ID (%s): %w", diffID, err)
 	}
 
 	layer, err := a.image.LayerByDiffID(h)
 	if err != nil {
-		return "", nil, xerrors.Errorf("failed to get the layer (%s): %w", diffID, err)
+		return "", 0, nil, xerrors.Errorf("failed to get the layer (%s): %w", diffID, err)
 	}
 
 	// digest is a hash of the compressed layer
 	var digest string
 	if a.isCompressed(layer) {
 		d, err := layer.Digest()
 		if err != nil {
-			return "", nil, xerrors.Errorf("failed to get the digest (%s): %w", diffID, err)
+			return "", 0, nil, xerrors.Errorf("failed to get the digest (%s): %w", diffID, err)
 		}
 		digest = d.String()
 	}
 
+	size, err := layer.Size()
+	if err != nil {
+		return "", 0, nil, xerrors.Errorf("failed to get the layer size (%s): %w", diffID, err)
+	}
+
 	rc, err := layer.Uncompressed()
 	if err != nil {
-		return "", nil, xerrors.Errorf("failed to get the layer content (%s): %w", diffID, err)
+		return "", 0, nil, xerrors.Errorf("failed to get the layer content (%s): %w", diffID, err)
 	}
-	return digest, rc, nil
+	return digest, size, rc, nil
```

Comment:
I think we can return the cached layer here. Also, we should always cache layers when `--slow` is passed.

Reply:
In addition, cachedFile is no longer needed.

Reply:
Done